src/test/java/htsjdk/samtools/cram/RANSInteropTest.java

```diff
@@ -0,0 +1,166 @@
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static htsjdk.samtools.cram.CRAMInteropTestUtils.filterEmbeddedNewlines;
+import static htsjdk.samtools.cram.CRAMInteropTestUtils.getInteropCompressedFilePaths;
+import static htsjdk.samtools.cram.CRAMInteropTestUtils.getParamsFormatFlags;
+import static htsjdk.samtools.cram.CRAMInteropTestUtils.getUnCompressedFilePath;
+
+/**
+ * RANSInteropTest tests if the htsjdk RANS4x8 and RANSNx16 implementations are interoperable
+ * with the htslib implementations. The test files for Interop tests is kept in a separate repository,
+ * currently at https://github.com/samtools/htscodecs so it can be shared across htslib/samtools/htsjdk.
+ *
+ * For local development env, the Interop test files must be downloaded locally and made available at "../htscodecs/tests"
+ * For CI env, the Interop test files are made available from the existing samtools installation
+ * at "/samtools-1.14/htslib-1.14/htscodecs/tests"
+ */
+public class RANSInteropTest extends HtsjdkTest {
+    public static final String COMPRESSED_RANS4X8_DIR = "r4x8";
+    public static final String COMPRESSED_RANSNX16_DIR = "r4x16";
+
+    // RANS4x8 codecs and testdata
+    public Object[][] get4x8TestCases() throws IOException {
+
+        // params:
+        // compressed testfile path, uncompressed testfile path,
+        // RANS encoder, RANS decoder, RANS params
+        final List<Object[]> testCases = new ArrayList<>();
+        for (Path path : getInteropCompressedFilePaths(COMPRESSED_RANS4X8_DIR)) {
+            Object[] objects = new Object[]{
+                    path,
+                    getUnCompressedFilePath(path),
+                    new RANS4x8Encode(),
+                    new RANS4x8Decode(),
+                    new RANS4x8Params(RANSParams.ORDER.fromInt(getParamsFormatFlags(path)))
+            };
+            testCases.add(objects);
+        }
+        return testCases.toArray(new Object[][]{});
+    }
+
+    // RANSNx16 codecs and testdata
+    public Object[][] getNx16TestCases() throws IOException {
+
+        // params:
+        // compressed testfile path, uncompressed testfile path,
+        // RANS encoder, RANS decoder, RANS params
+        final List<Object[]> testCases = new ArrayList<>();
+        for (Path path : getInteropCompressedFilePaths(COMPRESSED_RANSNX16_DIR)) {
+            Object[] objects = new Object[]{
+                    path,
+                    getUnCompressedFilePath(path),
+                    new RANSNx16Encode(),
+                    new RANSNx16Decode(),
+                    new RANSNx16Params(getParamsFormatFlags(path))
```

Comment:
The intention behind this change is to make sure that the RANS interop tests dynamically lists and uses all the test files available in the "dat/r4x8" and "dat/r4x16" directories, which were previously hardcoded. Instead of specifying various combinations of FormatFlags, we now automatically include all available files. 
It also accommodates new files added to the above mentioned directories in the future. Additionally, the util method "getParamsFormatFlags" introduced in this update can be reused for the InteropTesting of the other codecs.

Reply:
As per our conversation yesterday, I will add the previous Data Provider to the "testRANSRoundTrip" Test.